runtime/westend/src/weights/xcm/mod.rs

```diff
@@ -0,0 +1,191 @@
+mod pallet_xcm_benchmarks_fungible;
+mod pallet_xcm_benchmarks_generic;
+
+use crate::Runtime;
+use frame_support::weights::Weight;
+use sp_std::prelude::*;
+use xcm::{latest::prelude::*, DoubleEncoded};
+
+use pallet_xcm_benchmarks_fungible::WeightInfo as XcmBalancesWeight;
+use pallet_xcm_benchmarks_generic::WeightInfo as XcmGeneric;
+
+pub enum AssetTypes {
+	Balances,
+	Unknown,
+}
+
+impl From<&MultiAsset> for AssetTypes {
+	fn from(asset: &MultiAsset) -> Self {
+		match asset {
+			MultiAsset { id: Concrete(MultiLocation { parents: 0, interior: Here }), .. } =>
+				AssetTypes::Balances,
+			_ => AssetTypes::Unknown,
+		}
+	}
+}
+
+trait WeighMultiAssets {
+	fn weigh_multi_assets(&self, balances_weight: Weight) -> Weight;
+}
+
+// TODO wild case
+impl WeighMultiAssets for MultiAssetFilter {
+	fn weigh_multi_assets(&self, balances_weight: Weight) -> Weight {
+		match self {
+			Self::Definite(assets) => assets
+				.inner()
+				.into_iter()
+				.map(|m| <AssetTypes as From<&MultiAsset>>::from(m))
+				.map(|t| match t {
+					AssetTypes::Balances => balances_weight,
+					AssetTypes::Unknown => Weight::MAX,
+				})
+				.fold(0, |acc, x| acc.saturating_add(x)),
+			_ => Weight::MAX,
+		}
+	}
+}
+
+impl WeighMultiAssets for MultiAssets {
+	fn weigh_multi_assets(&self, balances_weight: Weight) -> Weight {
+		self.inner()
+			.into_iter()
+			.map(|m| <AssetTypes as From<&MultiAsset>>::from(m))
+			.map(|t| match t {
+				AssetTypes::Balances => balances_weight,
+				AssetTypes::Unknown => Weight::MAX,
+			})
+			.fold(0, |acc, x| acc.saturating_add(x))
+	}
+}
+
+pub struct WestendXcmWeight<Call>(core::marker::PhantomData<Call>);
+impl<Call> XcmWeightInfo<Call> for WestendXcmWeight<Call> {
+	fn withdraw_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::withdraw_asset())
+	}
+	fn reserve_asset_deposited(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmGeneric::<Runtime>::reserve_asset_deposited())
+	}
+	// TODO none of these need effects
+	fn receive_teleported_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::receive_teleported_asset())
+	}
+	fn query_response(_query_id: &u64, _response: &Response, max_weight: &u64) -> Weight {
+		*max_weight
+	}
+	fn transfer_asset(assets: &MultiAssets, _dest: &MultiLocation) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::transfer_asset())
+	}
+	fn transfer_reserve_asset(
+		assets: &MultiAssets,
+		_dest: &MultiLocation,
+		_xcm: &Xcm<()>,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::transfer_reserve_asset())
+	}
+	fn transact(
+		_origin_type: &OriginKind,
+		_require_weight_at_most: &u64,
+		_call: &DoubleEncoded<Call>,
+	) -> Weight {
+		XcmGeneric::<Runtime>::transact()
```

Comment:
Transact has no arguments? values seem also mock, what's the scenario of the generic xcm calls?

Reply:
Yeah this is a mock, and probably will change once i work on generic benchmark